libs/animation/animation.ts

```diff
@@ -1,13 +1,52 @@
+enum AnimationPath {
+    //% block="fly to center of screen"
+    FlyToCenter,
+    //% block="shake"
+    Shake,
+    //% block="bounce (right)"
+    BounceRight,
+    //% block="bounce (left)"
+    BounceLeft,
+    //% block="parachute (right)"
+    ParachuteRight,
+    //% block="parachute (left)"
+    ParachuteLeft,
+    //% block="ease (right)"
+    EaseRight,
+    //% block="ease (left)"
+    EaseLeft,
+    //% block="ease (down)"
+    EaseDown,
+    //% block="ease (up)"
+    EaseUp,
+    //% block="wave (right)"
+    WaveRight,
+    //% block="wave (left)"
+    WaveLeft,
+    //% block="bobbing (in place)"
+    Bobbing,
+    //% block="bobbing (right)"
+    BobbingRight,
+    //% block="bobbing (left)"
+    BobbingLeft
+}
+
 /*
     Animation library for sprites
 */
 //% color="#03AA74" weight=78 icon="\uf021"
 namespace animation {
-    //Handles all the updates
-    let animations: Animation[];
+    export interface AnimationState {
+        movementAnimation?: MovementAnimation,
+        imageAnimation?: ImageAnimation
+    }
+
+    // Stores the animations for the current scene
+    let animations: AnimationState;
```

Comment:
animations should be stored in the scene.state.

Reply:
Is scene.state a very new feature? I'm not seeing it anywhere.

Reply:
It is a property on the current scene (game.currentScene())

Reply:
I'm not seeing a state property there. Here's where I'm looking: https://github.com/microsoft/pxt-common-packages/blob/master/libs/game/scene.ts.

Are you suggesting I just add a property to the scene class to store animations?

Reply:
https://github.com/microsoft/pxt-common-packages/blob/master/libs/game/scene.ts#L163

Reply:
That property is read-only (no setter).

Reply:
You can did fields tonit. ...data[“dhrveihs”] = ...

Reply:
`game.currentScene().data["animationState"] = yourValue`